### PR TITLE
Collect type coverage metrics, add badge to README

### PR DESCRIPTION
**Pull request by morozov**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

Just another code quality metric which unlike code coverage should be green.

### CODE REVIEW

**Comment by morozov** [comment]
According to https://github.com/psalm/shepherd/issues/6#issuecomment-570797519, type coverage will be only collected once the `psalm` configuration gets merged into the default branch.

File: README.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-| [4.0-dev][4.0] | [3.0][3.0] | [2.12][2.12] |
+| [4.0-dev][4.0] | [3.0][3.0] | [2.13][2.13] |
 |:----------------:|:----------:|:----------:|
-| [![GitHub Actions][GA 4.0 image]][GA 4.0] | [![GitHub Actions][GA 3.0 image]][GA 3.0] | [![GitHub Actions][GA 2.12 image]][GA 2.12] |
-| [![AppVeyor][AppVeyor 4.0 image]][AppVeyor 4.0] | [![AppVeyor][AppVeyor 3.0 image]][AppVeyor 3.0] | [![AppVeyor][AppVeyor 2.12 image]][AppVeyor 2.12] |
-| [![Code Coverage][Coverage image]][CodeCov 4.0] | [![Code Coverage][Coverage 3.0 image]][CodeCov 3.0] | [![Code Coverage][Coverage 2.12 image]][CodeCov 2.12] |
+| [![GitHub Actions][GA 4.0 image]][GA 4.0] | [![GitHub Actions][GA 3.0 image]][GA 3.0] | [![GitHub Actions][GA 2.13 image]][GA 2.13] |
+| [![AppVeyor][AppVeyor 4.0 image]][AppVeyor 4.0] | [![AppVeyor][AppVeyor 3.0 image]][AppVeyor 3.0] | [![AppVeyor][AppVeyor 2.13 image]][AppVeyor 2.13] |
+| [![Code Coverage][Coverage image]][CodeCov 4.0] | [![Code Coverage][Coverage 3.0 image]][CodeCov 3.0] | [![Code Coverage][Coverage 2.13 image]][CodeCov 2.13] |
+| N/A | N/A | [![Code Coverage][TypeCov 2.13 image]][TypeCov 2.13] |
```
Comment (greg0ire) [inline README.md:8]: Why should it say N/A for more recent branches? Once this is merged up, that metric will be computed for these branches too, won't it?

Comment (morozov) [reply README.md:8]: No, as far as I understand shepherd currently supports badges only for the default branch. There's no documentation on how to display a badge for the given branch. I inspected their code and didn't find the support for branches.

**Review by greg0ire — APPROVED** [review]

**Comment by morozov** [comment]
Almost correct. I forgot that the default branch is `3.1.x`, not `2.13.x`.

I temporarily switched the default branch to `2.13.x`, manually triggered the static analysis build, and Shepherd [picked it up](https://shepherd.dev/github/doctrine/dbal). Just need to update the README to display the badge for `3.1.x` instead and merge the changes up.

**Pull request merged**